Source/MSBuildTools.Unity.NuGet/UnityMetaFileGenerator.targets

```diff
@@ -0,0 +1,200 @@
+      <UnityPlayerExclusionMask Include="$(ExcludedUnityPlayers)" />
+      <ExcludedUnityPlayer Include="@(UnityPlayerDefinition)" Condition=" '@(UnityPlayerExclusionMask)' == '@(UnityPlayerDefinition)' AND '%(Identity)' != '' " />
+      <PlatformExclusion Include="%(ExcludedUnityPlayer.Platforms)" />
+    </ItemGroup>
+
+    <!-- Generate a deterministic asset id. -->
+    <GenerateAssetId AssetName="%(RequiredUnityMetaFiles.Filename)$(UnityPlayer)">
+      <Output TaskParameter="AssetId" PropertyName="AssetId" />
+    </GenerateAssetId>
+
+    <!-- Write the standard .meta header. -->
+    <WriteLinesToFile File="$(MetaFilePath)" Lines="$([System.String]::Format($(UnityMetaFileHeader), '$(AssetId)'))" />
+
+    <!-- Write an entry in the .meta for each supported Unity player. -->
+    <WriteLinesToFile File="$(MetaFilePath)" Lines="@(PlatformInclusion->'$([System.String]::Format($(UnityMetaFilePlayerEnabledFormat), '%(PlatformInclusion.Identity)', '%(PlatformInclusion.IsEnabled)'))')" />
+
+    <!-- Write an entry for each excluded platform. -->
+    <WriteLinesToFile File="$(MetaFilePath)" Lines="@(PlatformExclusion->'$([System.String]::Format($(UnityMetaFilePlatformExcludedFormat), '%(PlatformExclusion.Identity)', '1'))')" />
+
+    <!-- Gather DLL specific MonoBehaviour execution order settings -->
+    <ItemGroup>
+      <DllScriptExecutionOrders Include="@(DllExecutionOrderOverrides)" Condition="'%(DllExecutionOrderOverrides.Identity)'=='$(DllName)'"/>
+      <MonoBehaviourExecutionOrders Include="%(DllScriptExecutionOrders.MonoBehaviourSpecificOverrides)" />
+    </ItemGroup>
+
+    <!-- Write an entry for each MonoBehaviour with a non-default execution order in this precompiled DLL -->
+    <WriteLinesToFile File="$(MetaFilePath)" Lines="%20%20executionOrder:" />
+    <WriteLinesToFile File="$(MetaFilePath)" Lines="@(MonoBehaviourExecutionOrders->'    %(Identity)')"/>
+
+    <!-- Add the generated .meta file to the FileWrites so it gets cleaned up properly and handles incremental builds. -->
+    <ItemGroup>
+      <FileWrites Include="$(MetaFilePath)" />
+    </ItemGroup>
+  </Target>
+
+  <!--Generates a Unity asset id (formatted guid) from an asset name (text).
+      Since Unity asset ids are guids (128 bit), use an MD5 hash (also 128 bits) formatted as a hex string. -->
+  <UsingTask TaskName="GenerateAssetId" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <AssetName ParameterType="System.String" Required="true" />
+      <AssetId ParameterType="System.String" Output="true" />
+    </ParameterGroup>
+    <Task>
+      <Using Namespace="System.Linq" />
+      <Using Namespace="System.Security.Cryptography" />
+      <Using Namespace="System.Text" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+            byte[] inputBytes = Encoding.UTF8.GetBytes(this.AssetName);
```

Comment:
Could we switch this to use the ProjectGuid instead, and increment for each player?

Reply:
We don't have a project guid in most cases. Most of these dlls will come from ProjectReferences and PackageReferences.

Reply:
Spoke offline.